src/server/pfs/cmds/cmds.go

```diff
@@ -503,74 +490,26 @@ $ {{alias}} foo@XXX -b bar@baz`,
 			}
 			defer c.Close()
 
-			w := tabwriter.NewWriter(os.Stdout, pretty.CommitHeader)
-			writeCommitInfo := func(ci *pfs.CommitInfo) error {
-				if raw {
-					return marshaller.Marshal(os.Stdout, ci)
-				}
-				pretty.PrintCommitInfo(w, ci, fullTimestamps)
-				return nil
-			}
-			defer func() {
-				if err := w.Flush(); retErr == nil {
-					retErr = err
-				}
-			}()
-
-			waitCommit := func(commit *pfs.Commit) error {
-				ci, err := c.WaitCommit(commit.Branch.Repo.Name, commit.Branch.Name, commit.ID)
-				if err != nil {
-					return err
-				}
-				return writeCommitInfo(ci)
+			commitInfo, err := c.WaitCommit(commit.Branch.Repo.Name, commit.Branch.Name, commit.ID)
+			if err != nil {
+				return err
 			}
 
-			waitBranches := func(commitsetID string) error {
-				for _, branch := range toBranches {
-					if err := waitCommit(client.NewCommit(branch.Repo.Name, branch.Name, commitsetID)); err != nil {
-						return err
-					}
-				}
-				return nil
+			if raw {
+				return cmdutil.Encoder(output, os.Stdout).EncodeProto(commitInfo)
+			} else if output != "" {
+				return errors.New("cannot set --output (-o) without --raw")
 			}
 
-			// There are separate codepaths for whether or not we are waiting on a
-			// commit or commitset and whether we are interested in all the commits in
-			// the set or only specific branches.
-			if commit != nil {
-				if len(toBranches) != 0 {
-					// We just want to resolve the commit to get the ID and wait on the downstream branches
-					ci, err := c.InspectCommit(commit.Branch.Repo.Name, commit.Branch.Name, commit.ID)
-					if err != nil {
-						return err
-					}
-					return waitBranches(ci.Commit.ID)
-				}
-				// We are just waiting on a single commit, block on it and exit
-				return waitCommit(commit)
-			} else {
-				if len(toBranches) == 0 {
-					return waitBranches(commitsetID)
-				}
-				// We are waiting for the entire commitset to finish
-				commitInfos, err := c.WaitCommitSetAll(commitsetID)
-				if err != nil {
-					return err
-				}
-				for _, commitInfo := range commitInfos {
-					if err := writeCommitInfo(commitInfo); err != nil {
-						return err
-					}
-				}
+			ci := &pretty.PrintableCommitInfo{
+				CommitInfo:     commitInfo,
+				FullTimestamps: fullTimestamps,
 			}
-
-			return nil
+			return pretty.PrintDetailedCommitInfo(os.Stdout, ci)
 		}),
 	}
-	waitCommit.Flags().VarP(&branches, "branch", "b", "Wait only for commits in the specified set of branches")
-	waitCommit.MarkFlagCustom("branch", "__pachctl_get_branch")
-	waitCommit.Flags().AddFlagSet(rawFlags)
-	waitCommit.Flags().AddFlagSet(fullTimestampsFlags)
+	waitCommit.Flags().AddFlagSet(outputFlags)
+	waitCommit.Flags().AddFlagSet(timestampFlags)
```

Comment:
should this register a completion function?

Reply:
I think we need to do a completion update in a separate pass, it's definitely out-of-sync with a lot of the changes in 2.0.